ports/esp32/modwifi_csi.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2025 Francesco Pace <francesco.pace@gmail.com>
+ * Copyright (c) 2025 MicroPython CSI Module Contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef MICROPY_INCLUDED_ESP32_MODWIFI_CSI_H
+#define MICROPY_INCLUDED_ESP32_MODWIFI_CSI_H
+
+#include "py/obj.h"
+
+#if MICROPY_PY_NETWORK_WLAN_CSI
+
+#include "esp_wifi.h"
+#include "esp_wifi_types.h"
+
+// Maximum CSI data size: ESP32 can provide up to 512 bytes
+#define CSI_MAX_DATA_LEN 512
+
+// Default buffer size (number of frames)
+#define CSI_DEFAULT_BUFFER_SIZE 16
```

Comment:
If this works with fewer than 16 buffers, I Wonder if this value should be able to be overridden in `mpconfigport.h`.